### PR TITLE
Feature/music commands

### LINKED ISSUE #206: Music commands
- [x] Play
- [x] Pause
- [x] Volume (premium)
- [x] Search (premium)
- [ ] Seek (premium) - Skip to a specific portion of the song.
- [x] Skip
- [x] Queue
- [x] Join
- [x] Leave
- [ ] Favorite (premium) - Needs DB implementation

### LINKED ISSUE #208: Custom fork of InteractivityAddon needs to be included in git.
Without it, nobody can clone and compile the code properly.

### PR DESCRIPTION
**Pull request by hburn7**
PR for all related music commands (will resolve #206)

### CODE REVIEW

**Comment by hburn7** [comment]
Note: This PR replaces the current Discord.InteractivityAddon dependency with a local DLL reference to my own fork of this lib. Not sure how this will play out when merging. @BitMasher

**Comment by hburn7** [comment]
@BitMasher I will need your help with getting "Lavalink", a java app, to run on each new docker deploy. https://github.com/Frederikam/Lavalink

File: Kaguya/Kaguya/Discord/Commands/Music/Leave.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            if (!_lavaNode.TryGetPlayer(Context.Guild, out var player))
+            {
+                await SendBasicErrorEmbedAsync("I couldn't find the music player for this server. Please ensure I am connected to a voice channel " +
+                                               "before using this command.");
+
+                return;
+            }
+
+            await SendBasicSuccessEmbedAsync($"Disconnected from {player.VoiceChannel.Name.AsBold()}.");
+            await _lavaNode.LeaveAsync(player.VoiceChannel);
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/Commands/Music/Leave.cs]: don't tell the user you've disconnected before you have actually disconnected

File: Kaguya/Kaguya/Discord/Commands/Music/Skip.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                var embed = GetBasicSuccessEmbedBuilder($"Skipped {curTrack.Title.AsBold()}").Build();
+                _interactivityService.SendEmbedWithDeletion(Context, embed, TimeSpan.FromSeconds(15));
+                
+                _interactivityService.SendEmbedWithDeletion(Context, MusicEmbeds.GetNowPlayingEmbedForTrack(player.Track), TimeSpan.FromSeconds(15));
+            }
+            else
+            {
+                if (skipCount.Value < 2)
+                {
+                    await SendBasicErrorEmbedAsync("If you are specifiying a skip count, you must skip 2+ tracks.");
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/Commands/Music/Skip.cs:78]: why is this necessary? wouldn't it be better to just use the same logic as line 67-72

File: Kaguya/Kaguya/Discord/Commands/Music/Skip.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+                    return;
+                }
+
+                int actualSkipCount = 0;
+                for (int i = 0; i < skipCount.Value; i++)
```
Comment (BitMasher) [inline Kaguya/Kaguya/Discord/Commands/Music/Skip.cs:84]: should be skipCount.Value-1, since the first skip is always the current song, otherwise this will always skip 1 extra song

File: Kaguya/Kaguya/Internal/Music/AudioService.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            {
+                _ = InitiateDisconnectAsync(args.Player, TimeSpan.FromSeconds(10));
+
+                return;
+            }
+
+            if (!(queueable is LavaTrack track))
+            {
+                return;
+            }
```
Comment (BitMasher) [inline Kaguya/Kaguya/Internal/Music/AudioService.cs]: why wouldn't this be in a loop until either a playable track is retrieved or the queue is empty?

File: Kaguya/Kaguya/Internal/Music/AudioService.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                value = new CancellationTokenSource();
+                _disconnectTokens.TryAdd(player.VoiceChannel.Id, value);
+            }
+            else if (value.IsCancellationRequested)
+            {
+                _disconnectTokens.TryUpdate(player.VoiceChannel.Id, new CancellationTokenSource(), value);
+                value = _disconnectTokens[player.VoiceChannel.Id];
+            }
+
+            bool isCancelled = SpinWait.SpinUntil(() => value.IsCancellationRequested, timeSpan);
```
Comment (BitMasher) [inline Kaguya/Kaguya/Internal/Music/AudioService.cs]: why would you use SpinWait instead of Task.Delay? SpinWait is synchronous

File: Kaguya/Kaguya/Internal/Music/MusicEmbeds.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            if (queue.Count == 0)
+            {
+                descSb.AppendLine("No tracks enqueued.".AsItalics());
+            }
+
+            for (int i = 0; i < queue.Count; i++)
+            {
+                LavaTrack track = queue.ElementAt(i);
+                descSb.AppendLine($"#{i + 1}. {track.Title.AsBold()} - {track.Author} ({track.Duration:mm\\:ss})");
+            }
```
Comment (BitMasher) [inline Kaguya/Kaguya/Internal/Music/MusicEmbeds.cs]: this could be cleaner using a foreach statement

**Review by BitMasher — CHANGES_REQUESTED** [review]

File: Kaguya/Kaguya/Internal/Music/AudioService.cs (hunk 5 above)
Comment (hburn7) [reply Kaguya/Kaguya/Internal/Music/AudioService.cs]: Hmm. This was in the example provided here:

https://victoria.yucked.wtf/guides/samples/auto-disconnect.html

File: Kaguya/Kaguya/Internal/Music/AudioService.cs (hunk 4 above)
Comment (hburn7) [reply Kaguya/Kaguya/Internal/Music/AudioService.cs]: If `player.Queue.TryDequeue(out LavaTrack queueable)` equals true, that means a successful track is able to be received and the queue is not empty, otherwise the opposite is the case (empty queue).

File: Kaguya/Kaguya/Discord/Commands/Music/Skip.cs (hunk 3 above)
Comment (hburn7) [reply Kaguya/Kaguya/Discord/Commands/Music/Skip.cs:84]: From my testing, the current for loop does function correcty
